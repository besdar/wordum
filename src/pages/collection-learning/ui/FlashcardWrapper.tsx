import React from 'react';
import {Grid} from '../../../shared/ui/Grid';
import {FlashcardContainer} from './FlashcardContainer';

type Props = {
  children: React.ReactNode;
  footer: React.ReactNode;
  header: React.ReactNode;
};

export const FlashcardWrapper = ({children, footer, header}: Props) => (
  <Grid direction="column" fillAwailableSpace rowGap={20}>
    {header}
    <FlashcardContainer>{children}</FlashcardContainer>
    <Grid alignItems="center" justifyContent="center">
      {footer}
    </Grid>
  </Grid>
);
