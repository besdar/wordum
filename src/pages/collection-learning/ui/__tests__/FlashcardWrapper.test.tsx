import {render} from '@testing-library/react-native';
import {FlashcardWrapper} from '../FlashcardWrapper';
import {View} from 'react-native';
import React from 'react';

describe('FlashcardWrapper', () => {
  it('should match the snapshot', () => {
    const {toJSON} = render(
      <FlashcardWrapper header={<View />} footer={<View />}>
        <View />
      </FlashcardWrapper>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
