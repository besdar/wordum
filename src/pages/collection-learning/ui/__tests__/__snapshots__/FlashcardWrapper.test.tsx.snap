// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardWrapper should match the snapshot 1`] = `
<View
  style={
    [
      {
        "display": "flex",
        "width": "100%",
      },
      {
        "alignItems": undefined,
        "columnGap": undefined,
        "flexBasis": undefined,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "gap": undefined,
        "justifyContent": "flex-start",
        "padding": undefined,
        "rowGap": 20,
      },
      {
        "flexGrow": 1,
      },
    ]
  }
>
  <View />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgb(247, 243, 249)",
        "borderRadius": 10,
        "flexGrow": 1,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
        "width": "100%",
      }
    }
    testID="surface-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(247, 243, 249)",
          "borderColor": undefined,
          "borderRadius": 10,
          "columnGap": undefined,
          "display": "flex",
          "flex": undefined,
          "flexBasis": undefined,
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "gap": undefined,
          "justifyContent": "center",
          "padding": undefined,
          "rowGap": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
      testID="surface"
    >
      <View />
    </View>
  </View>
  <View
    style={
      [
        {
          "display": "flex",
          "width": "100%",
        },
        {
          "alignItems": "center",
          "columnGap": undefined,
          "flexBasis": undefined,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "gap": undefined,
          "justifyContent": "center",
          "padding": undefined,
          "rowGap": undefined,
        },
      ]
    }
  >
    <View />
  </View>
</View>
`;
